spec/features/new_results_page_filters/subjects_spec.rb

```diff
@@ -0,0 +1,329 @@
+require 'rails_helper'
+
+RSpec.feature 'Results page new subject filter' do
+  include StubbedRequests::Courses
+  include StubbedRequests::SubjectAreas
+  include StubbedRequests::Subjects
+
+  let(:filter_page) { PageObjects::Page::ResultFilters::SubjectPage.new }
+  let(:results_page) { PageObjects::Page::ResultsWithNewFilters.new }
+  let(:base_parameters) { results_page_parameters }
+
+  before do
+    activate_feature(:new_filters)
+
+    stub_courses(query: base_parameters, course_count: 10)
+    stub_subject_areas
+    stub_subjects
+  end
+
+  describe 'applying a filter' do
+    before do
+      stub_courses(query: base_parameters, course_count: 10)
+    end
+
+    context 'with less than 4 subjects selected' do
+      before do
+        stub_courses(
+          query: base_parameters.merge('filter[subjects]' => '00,F1'),
+          course_count: 10,
+        )
+      end
+
+      it 'lists the results' do
+        results_page.load
+        results_page.subjects_filter.link.click
+
+        expect(filter_page.heading.text).to eq(I18n.t('page_titles.subjects_filter'))
+        filter_page.subject_areas.first.subjects[0].checkbox.click
+        filter_page.subject_areas.second.subjects[3].checkbox.click
+
+        filter_page.continue.click
+
+        expect(results_page.heading.text).to eq('Teacher training courses')
+        expect(results_page.subjects_filter.subjects.first.text).to eq('Chemistry, Primary')
+
+        expect(results_page.courses.count).to eq(10)
+      end
+    end
+
+    context 'with subjects selected' do
+      before do
+        stub_courses(
+          query: base_parameters.merge('filter[subjects]' => '00,01,F1,Q8,P3'),
+          course_count: 10,
+        )
+      end
+
+      it 'lists the results' do
+        results_page.load
+        results_page.subjects_filter.link.click
+
+        expect(filter_page.heading.text).to eq(I18n.t('page_titles.subjects_filter'))
+        filter_page.subject_areas.first.subjects[0].checkbox.click
+        filter_page.subject_areas.first.subjects[1].checkbox.click
+        filter_page.subject_areas.second.subjects[3].checkbox.click
+        filter_page.subject_areas.second.subjects[5].checkbox.click
+        filter_page.subject_areas.second.subjects[6].checkbox.click
+
+        filter_page.continue.click
+
+        expect(results_page.heading.text).to eq('Teacher training courses')
+        expect(results_page.subjects_filter.subjects.first.text).to eq(
+          'Chemistry, Classics, Communication and media studies, Primary, Primary with English',
+        )
+
+        expect(results_page.courses.count).to eq(10)
+      end
+    end
+  end
+
+  context 'with only SEND courses selected' do
```

Comment:
Again i might be missing something, but I'm not 100% sure why we're testing this. Is SEND a subject on the backend and we're checking it's been filtered out at some point?

Reply:
Again my bad - missing an assertion, have fixed. When  a candidate goes to the subjects filter page and selects SEND courses we want to make sure the SEND checkbox is ticked on the results page.